### ISSUE TITLE
Resizing packery gallery via angular 

### ISSUE THREAD

**Issue by pliablepixels** [metafizzy/packery#349]
Hi there, I am trying to combine the benefit of angular's expression to resize a packery gallery and am facing issues on how the reflow happens. Basically, if I dynamically change the css (grid-item-20/40/60 etc) and invoke pckry reloadItems and layout the layout doesn't work.

An example is below
http://codepen.io/pliablepixels/pen/VaYpQj

I've even tried to delete the pckry object each time and re-create - that do did not work. Can you take a look at what I am doing wrong? I tried to jump into the code but its a bit ahead of my knowledge

thx

**Comment by desandro** [comment]
This should work

``` js
  $scope.sliderChanged = function(val)
  {
    $scope.myscale = "grid-item-"+($scope.slider.val*10).toString();
    pckry.layout();
  };
```

But the change has not been immediately made. I'm not sure if this is standard Angular behavior or what. Hacking it with `setTimeout` shows how it should work

``` js
  $scope.sliderChanged = function(val)
  {
    $scope.myscale = "grid-item-"+($scope.slider.val*10).toString();
    console.log( pckry.items[0].element.className )
    setTimeout( function() {
      pckry.layout();
    })
  };
```

I also added `$scale` to the `.grid-sizer`, since its size needs to be updated as well.

See demo http://codepen.io/desandro/pen/a7aca75444c6f29efc29a4bef7db889c/

**Comment by pliablepixels** [comment]
thank you very much @desandro ! The grid-sizer size was the missing link. However, my next plan is to allow for individual items to be sized differently. So in the case where we have 5 images, each with a different % size, what would be the grid-sizer value?

**Comment by desandro** [comment]
It should be a multiple of the items width. So if you have items with 20%, 40%, and 60% width, `.grid-sizer` should be 20%.

**Comment by pliablepixels** [comment]
Thank you - so if I understand this right, lets say I have the following setup:

Image 1: 60%, Image 2: 10%, Image 3:40%, Image 4: 80% then grid-sizer should be 10% ? i.e. grid-size = minimum of width of all images (all images are in %)

**Comment by pliablepixels** [comment]
@desandro please take a look at the modified codepen - http://codepen.io/pliablepixels/pen/ZWbqJZ
This is where I finally want to land up. You can drag the slider to scale all images or tap each image to only scale that http://codepen.io/pliablepixels/pen/ZWbqJZ

I'd appreciate you taking a look and letting me know what I need to do to make it right - the layout seems to be going all over the place now.

**Comment by desandro** [comment]
yes. If grid items can be 20%, 30%, 40%, 50%, etc, then the largest common multiple would be 10%. See http://codepen.io/desandro/pen/4286c23b0ceea14af9cdd5ff49c65faf/

**Comment by pliablepixels** [comment]
Excellent - my progress so far 
http://pliablepixels.github.io/2016/03/07/Montage-Awesomeness.html

I'm using your library to re-arrange video feeds.  I've integrated it with the load and save functions (I've extended it to also save the size) here #337 and it works well.

I'm still struggling with the odd case of packery layouts getting messed up - not sure if its angular or not - every once in a while the layout gets spaced out very wide. Anyway, thanks a bunch!
